smart_open/s3.py

```diff
@@ -663,9 +663,10 @@ def seek(self, offset, whence=constants.WHENCE_START):
             whence = constants.WHENCE_START
             offset += self._current_pos
 
-        self._current_pos = self._raw_reader.seek(offset, whence)
+        if not (whence == constants.WHENCE_START and offset == self._current_pos):
```

Comment:
```suggestion
        if not (whence == constants.WHENCE_START and offset == self._current_pos):
```

The optimization idea is good, but we need to maintain the current behavior of the library. That is, the seek should always be performed when opening the file. Perhaps we could initialize self._current_pos to -1 to achieve this.

Reply:
🦗